### PR TITLE
feat: new wallpaper viewer design

### PR DESCRIPTION
**Pull request by SuhasDissa**
## New Features
- Redesigned wallpaper view screen
- Redesigned wallpaper info dialog
- Show wallpapers in pager view (with horizontal scroll)
- More responsive filter editor
- Option to crop and apply wallpaper through the system gallery app (or any other app that supports it)

Please let me know if you have any suggestions

### CODE REVIEW

**Review by M00NJ — APPROVED** [review]

**Review by Bnyro — COMMENTED** [review]
> One thing: I'd prefer to keep the floating action bar instead of placing the buttons at the bottom with a transparent background, I feel like the current floating action bar with buttons make it look more Material Design 3.
Though I'm fine with the color palette placement changes, they no longer need to be part of that action bar (as in your changes).
Awesome apart from that 👍
>
(From Matrix)

**Comment by SuhasDissa** [comment]
>  One thing: I'd prefer to keep the floating action bar instead of placing the buttons at the bottom with a transparent background, I feel like the current floating action bar with buttons make it look more Material Design 3.

Floating Action Bar will look good. but we'll have to remove the blur effects to make it look less odd.

**Comment by SuhasDissa** [comment]
Should I leave the top bar transparent?

File: app/src/main/java/com/bnyro/wallpaper/ui/components/WallpaperView.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .zoomArea(zoomState),
+            contentAlignment = Alignment.Center
+        ) {
+            val lowRes = rememberAsyncImagePainter(model = wallpaper.thumb ?: wallpaper.imgSrc)
+            AsyncImage(
+                model = wallpaper.imgSrc,
+                contentDescription = "Wallpaper",
```
Comment (Bnyro) [inline app/src/main/java/com/bnyro/wallpaper/ui/components/WallpaperView.kt:114]: nit: please make it translatable

File: app/src/main/java/com/bnyro/wallpaper/ui/components/WallpaperView.kt [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            ) {
+                AnimatedVisibility(
+                    visible = showUi
+                ) {
+                    WallpaperViewTopBar(
+                        onClickBack = onClickBack,
+                        onClickInfo = {
+                            showInfoSheet = true
+                        },
+                        title = wallpaper.title ?: "Wallpaper"
```
Comment (Bnyro) [inline app/src/main/java/com/bnyro/wallpaper/ui/components/WallpaperView.kt:134]: nit: please make it translatable

**Comment by Bnyro** [comment]
> Should I leave the top bar transparent?
>
One second, I'll quickly build and test what looks better.

**Comment by Bnyro** [comment]
> Should I leave the top bar transparent?
>
I've tested both, I think it's better to keep it as it is now (top bar being *not* transparent)

**Comment by Bnyro** [comment]
Only one small UI nitpick left from my side, then I'm fully happy with this PR :)

The bottom sheet to edit the wallpaper currently has two filled buttons: "Reset" and "Apply".

My suggestion would be to place
1. the "Reset" button at the very left
2. the "Apply" button to the very right
3. a "Dismiss" button to the left of the "Apply" button

Apart from that I think that `TextButton`'s would look a bit nicer here than normal `Button`'s because they look less clunky.

**Review by Bnyro — APPROVED** [review]
Awesome, thanks for your hard work on that!

**Comment by SuhasDissa** [comment]
> Awesome, thanks for your hard work on that!

Thank you, I will probably create a follow up PR adding navigation support. Using dialogs for everything is not a good idea IMO.

**Pull request merged**